src/net_processing.cpp

```diff
@@ -488,6 +488,14 @@ struct Peer {
     /** Whether this peer should be disconnected and marked as discouraged (unless it has the noban permission). */
     bool m_should_discourage GUARDED_BY(m_misbehavior_mutex){false};
 
+    /** Protects block inventory data members */
+    RecursiveMutex m_block_inv_mutex;
```

Comment:
e04fb9a2df9f4d77c587bd25c608ce806c97ea45
Could using of `RecursiveMutex` type be avoided?
(as there was `Mutex cs_inventory;`)

Reply:
Good catch! `cs_inventory` was a RecursiveMutex when I opened this PR, and I missed the change when I rebased. Fixed.

Reply:
#19347 is yours, btw :)

Reply:
:flushed: